javaparser-core/src/main/javacc/java.jj

```diff
@@ -330,6 +330,7 @@ TOKEN :
 | < PRIVATE: "private" >
 | < PROTECTED: "protected" >
 | < PUBLIC: "public" >
+| < RECORD: "record" >
```

Comment:
Maybe you've already done this, but make sure "record" can be used as an identifier, and make sure the token type is identifier when that is the case (and record when it isn't.)